flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/connection/SimpleJdbcConnectionProvider.java

```diff
@@ -100,7 +100,16 @@ private Driver getDriver() throws SQLException, ClassNotFoundException {
 
     @Override
     public Connection getOrEstablishConnection() throws SQLException, ClassNotFoundException {
-        if (connection == null) {
+        if (connection != null) {
+            return connection;
+        }
+        if (jdbcOptions.getDriverName() == null) {
+            connection =
+                    DriverManager.getConnection(
+                            jdbcOptions.getDbURL(),
+                            jdbcOptions.getUsername().orElse(null),
+                            jdbcOptions.getPassword().orElse(null));
+        } else {
```

Comment:
What about changing `getDriver()` to `getDriver(String driverName)` and check not null in the method? This can avoid `getDriver()` been used accidentally.

Reply:
Hi @wuchong, I pushed a fixup commit with following work:
* Separate a static method `loadDriver(String driverName)` to load driver and do not-null checking there.
* Keep signature of old no-static `getDriver`(renamed to `getLoadedDriver`) unchanged.

I think a `driverName` parameter in no-static method might give maintainers illusion that it could be used to load a different driver while it is not.

Please check whether it solves your concern